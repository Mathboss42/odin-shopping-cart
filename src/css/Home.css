@font-face {
    font-family: 'ChakraPetch';
    src: url('./ChakraPetch-Bold.ttf');
    font-style: normal;
}

@font-face {
    font-family: 'Roboto';
    src: url('./Roboto-Medium.ttf');
    font-weight: 700;
    font-style: normal;
}

@font-face {
    font-family: 'Roboto-Bold';
    src: url('./Roboto-Bold.ttf');
    font-style: normal;
}

@font-face {
    font-family: 'Cyberpunk';
    src: url('./Cyberpunk.ttf');
}

:root {
    --cyber-blue: rgb(0, 240, 255);
    --cyber-deep-blue: rgb(0, 100, 107);
    --cyber-yellow: rgb(252, 238, 10);
    --cyber-deep-yellow: rgb(110, 105, 0);
    --cyber-deep-dark: black;
    --cyber-mid-dark: rgb(45, 55, 59);
}

* {
    padding: 0;
    margin: 0;
    box-sizing: border-box;
    color: whitesmoke;
    font-family: ChakraPetch, 'Roboto', sans-serif;
}

html, body {
    background-color:var(--cyber-deep-dark);
    height: 100%;
    overflow: auto;
}

.nav {
    user-select: none;
    margin-bottom: 50px;
}

h1 {
    font-family: 'Cyberpunk', sans-serif;
    color: var(--cyber-yellow);
    text-shadow: 2px 2px 0px var(--cyber-blue);
    font-size: 3rem;
    transform: translate(0, -5%);
}

.small-number {
    text-shadow: none;
    color: var(--cyber-blue);
    font-family: sans-serif;
    letter-spacing: -1px;
    font-weight: 100;
    font-size: x-large;
    font-style: italic;
    transform: translate(-35%, 70%);
}

.nav ul {
    display: flex;
    justify-content: space-around;
    list-style: none;
    padding: 24px;
    font-size: x-large;
}

.nav h1 {
    display: flex;
    justify-content: center;
    padding: 24px;
}

a {
    text-decoration: none;
    color: var(--cyber-blue);
}

li.selected {
    border: 1px solid var(--cyber-blue);
}

li {
    border: 1px solid transparent;
    padding: 0 6px;
}

li:hover > * {
    color: white;
}

li:hover.selected {
    border-color: white;
}

#root, .content-container {
    height: 100%;
    overflow: auto;
}

.content-container {
    padding: 50px;
}

.terminal {
    border: 1px solid var(--cyber-blue);
    padding: 50px;
}

.store-container.open-cart {
    display: grid;
    grid-template-columns: 2fr 1fr;
    gap: 20px;
}

.store-container {
    position: relative;
}

#show-cart-button {
    position: absolute;
    top: 0;
    left: 90%;
    /* transform: translate(-100%); */
}

h3 {
    font-size: 32px;
}

p {
    font-size: 24px;
    text-align: justify;
    text-justify: inter-word;
}

.blocks-container {
    display: flex;
    flex-direction: column;
    gap: 50px;
}

.text-block {
    width: min(100%, 600px);
    padding-bottom: 24px;
    border-bottom: 1px solid var(--cyber-yellow);
}

.text-block:nth-child(2n) {
    align-self: flex-end;
}

.serial {
    text-align: end;
    color: var(--cyber-blue);
    transform: translate(45px, 50px);
}

button {
    background-color: var(--cyber-yellow);
	border: none;
	padding: 0px 10px;
	font-size: 26px;
	cursor: pointer;
}

button > div, button {
	color: var(--cyber-deep-blue);
    transform: translate(0, -5%);
}

.items-container {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.item p {
    font-size: 20px;
}

.item h4 {
    font-size: 24px;
}

.item-category h3 {
    margin-bottom: 24px;
}

.item {
    padding: 8px;
    margin-bottom: 24px;
    border-bottom: 1px solid var(--cyber-yellow);
    border-left: 1px solid var(--cyber-yellow);
}

.item {
    width: min(100%, 800px);
    display: flex;
}

.item-right {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
}

.item-controls {
    align-self: flex-end;
}

img {
    display: block;
    width: min(40%, 192px);
    height: min(40%, 192px);
}

.cart {
    background-color: var(--cyber-yellow);
    padding: 16px;
}

.cart * {
    color: var(--cyber-deep-blue);
}

.cart button {
    border: 1px solid var(--cyber-deep-blue);
}

.cart-header {
    display: flex;
    justify-content: space-between;
}