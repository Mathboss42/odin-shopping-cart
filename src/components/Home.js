import '../css/Home.css';
import Nav from './Nav';
import Terminal from './Terminal';

function Home() {
    return (
        <div className="home content-container">
            <Nav selected={"Home"} />
            <Terminal>
                <div>nothing</div>
            </Terminal>
        </div>
    );
}

export default Home;
