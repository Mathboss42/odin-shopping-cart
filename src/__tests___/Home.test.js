import { render, screen } from '@testing-library/react';
import Home from '../components/Home';