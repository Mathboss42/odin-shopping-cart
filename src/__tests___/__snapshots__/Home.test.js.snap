// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Elements Contains three blocks of text inside terminal 1`] = `
<div>
  <div
    class="home content-container"
  >
    <div
      class="nav"
    >
      <h1>
        CyberShop
        <span
          class="small-number"
        >
          2023
        </span>
      </h1>
      <ul>
        <li
          class="selected"
        >
          <a
            href="/home"
          >
            HOME
          </a>
        </li>
        <li>
          <a
            href="/store"
          >
            STORE
          </a>
        </li>
      </ul>
    </div>
    <div
      class="terminal"
    >
      <div
        class="blocks-container"
      >
        <div
          class="text-block"
          id="description"
        >
          <h3>
            DESCRIPTION
          </h3>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Similique vero mollitia, aliquid nostrum autem fugiat, enim iure deleniti veniam maxime, sequi atque tempora ut! Culpa?
          </p>
        </div>
        <div
          class="text-block"
          id="values"
        >
          <h3>
            VALUES
          </h3>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Similique vero mollitia, aliquid nostrum autem fugiat, enim iure deleniti veniam maxime, sequi atque tempora ut! Culpa? Lorem ipsum dolor, sit amet consectetur adipisicing elit. Libero, perferendis.
          </p>
        </div>
        <div
          class="text-block"
          id="how"
        >
          <h3>
            HOW
          </h3>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Similique vero mollitia, aliquid nostrum autem fugiat, enim iure deleniti veniam maxime, sequi atque tempora ut! Culpa? Lorem ipsum dolor, sit amet consectetur adipisicing elit. Libero, perferendis.
          </p>
        </div>
      </div>
      <div
        class="serial"
      >
        16591B-77:a
      </div>
    </div>
  </div>
</div>
`;
